### PR TITLE
Add ImageDecoder operator, selecting implementation based on device argument

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

### CODE REVIEW

File: dali/pipeline/operators/decoder/decoder.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace dali {
+
+DALI_SCHEMA(Decoder)
+  .DocStr(R"code(Decode images. Implementation will be `nvJPEGDecoder` or `HostDecoder`
+depending on the selected backend (`mixed` and `cpu` respectively).
+Output of the decoder is in `HWC` ordering.)code")
+  .NumInput(1)
+  .NumOutput(1)
+  .AddOptionalArg("output_type",
```
Comment (JanuszL) [inline dali/pipeline/operators/decoder/decoder.cc]: What about this extra args that nvJPEG accepts?

Comment (JanuszL) [reply dali/pipeline/operators/decoder/decoder.cc]: And slice and crop, and split option for nvJPEG?

Comment (Kh4L) [reply dali/pipeline/operators/decoder/decoder.cc]: Please add `.AddParent("nvJPEGDecoder");`

Comment (jantonguirao) [reply dali/pipeline/operators/decoder/decoder.cc]: Solved

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by JanuszL** [comment]
I would do a followup and replace usage of HostDecoder and nvJPEG in our examples.

**Comment by mzient** [comment]
What about other python tests?

**Comment by jantonguirao** [comment]
> What about other python tests?

I'd rather update the other tests (changing HostDecoder/nvJPEGDecoder to ImageDecoder) in a separate PR

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Pull request merged**